src/utils/stackTrace.ts

```diff
@@ -87,6 +90,13 @@ export function captureStackTrace(): ParsedStackTrace {
   return { frames, frameTexts, apiName };
 }
 
+function isFilePartOfPlaywright(isTesting: boolean, fileName: string): boolean {
+  const isPlaywrightTest = fileName.includes(runnerNpmPkgLib);
```

Comment:
Why this check? What are we trying to achieve? Perhaps all this logic should be replaced by "fileName is not in <playwright>/lib/client"?

Reply:
This complexity was introduced in #7347, so let's ask Pavel.

Reply:
We can try simplifying it. The goal here is to identify the point where we hit Playwright Library API, treating @lpaywright/test as user land. As far as I remember, the complexity is around behaving differently while under test and while in production.